test/cqlpy/test_name.py

```diff
@@ -37,63 +45,196 @@ def passes_or_raises(expected_exception, match=None):
+# some other error.
+def test_table_name_length_gt_the_scylla_limit(cql, test_keyspace):
+    name = padded_name(MAX_NAME_LENGTH + 1)
+    with passes_or_raises(InvalidRequest, match=name):
+        with new_named_test_table(cql, test_keyspace, name, SCHEMA):
+            pass
+
+
+# Verifies that attempting to create a table with a name significantly longer than the limit
+# (e.g., 500 characters—which exceeds the Linux filesystem limit of 255 characters)
+# is gracefully rejected with an InvalidRequest exception.
+# This test is disabled for Cassandra because it causes a hang
+# (see CASSANDRA-20425 and CASSANDRA-20389).
 def test_table_name_length_500(cql, test_keyspace, cassandra_bug):
-    schema = 'p int, c int, PRIMARY KEY (p, c)'
-    n = padded_name(500)
-    with passes_or_raises(InvalidRequest, match=n):
-        with new_named_table(cql, test_keyspace, n, schema) as table:
+    name = padded_name(500)
+    with pytest.raises(InvalidRequest, match=name):
+        with new_named_test_table(cql, test_keyspace, name, SCHEMA):
+            pass
+
+
+# Verifies that a table name of exactly 207 characters is accepted when CDC is enabled.
+def test_table_name_length_eq_scylla_limit_and_cdc_enabled(cql):
+    with new_test_keyspace(
+        cql, "WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor': 1}"
+    ) as keyspace:
+        with new_named_test_table(
+            cql,
+            keyspace,
+            padded_name(MAX_NAME_LENGTH),
+            SCHEMA,
+            extra="with cdc = {'enabled': true}",
+        ):
+            pass
+
+
+# Verifies that a keyspace name of exactly 207 characters is accepted.
+def test_keyspace_name_length_eq_scylla_limit(cql):
+    with new_test_keyspace(
+        cql,
+        "WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor': 1}",
+        name=padded_name(MAX_NAME_LENGTH),
+    ):
+        pass
+
+
+# Verifies that a keyspace name longer than 207 characters is accepted (Cassandra 4/5)
+# or rejected with an InvalidRequest exception (Scylla/Cassandra 3).
+# The error message must contain the name of the keyspace, so that we can
+# verify that the test is actually testing the name length limit, and not
+# some other error.
+def test_keyspace_name_length_gt_than_scylla_limit(cql):
+    name = padded_name(MAX_NAME_LENGTH + 1)
+    with passes_or_raises(InvalidRequest, match=name):
+        with new_test_keyspace(
+            cql,
+            "WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor': 1}",
+            name=name,
+        ):
+            pass
+
+
+# Verifies that a materialized view name of exactly 207 characters is accepted.
+def test_mv_name_length_eq_scylla_limit(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, SCHEMA) as table:
+        pass
+        with new_materialized_view(
+            cql,
+            table,
+            "*",
+            "p, x",
+            "p is not null and x is not null",
```

Comment:
Again, I find all those separate lines excessive and just obscures what this test does with a spotlight on non-interesting details.